src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -229,130 +230,215 @@ public String getUserAnalysisOutputsPage(Model model) {
 	 *
 	 * @param submissionId the ID of the submission
 	 * @param model        Model for the view
+	 * @param principal    Principal {@link User}
 	 * @param locale       User's locale
 	 * @return name of the details page view
 	 */
+
 	@RequestMapping(value = "/{submissionId}", produces = MediaType.TEXT_HTML_VALUE)
-	public String getDetailsPage(@PathVariable Long submissionId, Model model, Locale locale) {
+	public String getDetailsPage(@PathVariable Long submissionId, Model model, final Principal principal,
+			Locale locale) {
 		logger.trace("reading analysis submission " + submissionId);
 		AnalysisSubmission submission = analysisSubmissionService.read(submissionId);
 		model.addAttribute("analysisSubmission", submission);
 
-		boolean canShareToSamples = false;
-		if (submission.getAnalysis() != null) {
-			canShareToSamples = analysisSubmissionSampleProcessor
-					.hasRegisteredAnalysisSampleUpdater(submission.getAnalysis().getAnalysisType());
+		final User currentUser = userService.getUserByUsername(principal.getName());
+
+		if(currentUser != null) {
```

Comment:
Shouldn't something more serious happen if the current user is `null`?

Reply:
I don't actually think that state can be reach here.

Reply:
Left a comment for this block of code. The check is only required for the AnalysisControllerTest

Reply:
Fixed in 9c4e640